### PR TITLE
Support for results with indirect route paths

### PR DESCRIPTION
**Pull request by dstiliadis**
Enables graph results to show multi-hop paths over indirection.

### CODE REVIEW

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "2d228557281d291791a81e2cce169bcde59781ed",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "60e08395f9089642508e9e2f3845571290b5cf4e",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "60e08395f9089642508e9e2f3845571290b5cf4e",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "1122ec18a0633b487dfcd9610cd77bb059a1887f",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "60e08395f9089642508e9e2f3845571290b5cf4e",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "1122ec18a0633b487dfcd9610cd77bb059a1887f",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "60e08395f9089642508e9e2f3845571290b5cf4e",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "2f94dc21df39c35a2dd11cbe22a87b465811884d",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "60e08395f9089642508e9e2f3845571290b5cf4e",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "3b9e126126d0a6797e2679935e6212316e8679e8",
    "pipeline": "master"
  }
]
```

File: specs/cloudnetworkquerydestination.spec [hunk 1]
```diff
@@ -11,6 +11,21 @@ model:
...
 attributes:
   v1:
+  - name: indirect
+    description: Returns true if this is an indirect path through an forwarding entities.
+    type: boolean
+    exposed: true
+    read_only: true
+    autogenerated: true
+
+  - name: indirectNode
```
Comment (primalmotion) [inline specs/cloudnetworkquerydestination.spec:21]: looks like it should be named `indirectNodeID`

File: specs/cloudnetworkquerydestination.spec [hunk 2]
```diff
@@ -11,6 +11,21 @@ model:
 # Attributes
 attributes:
   v1:
+  - name: indirect
```
Comment (primalmotion) [inline specs/cloudnetworkquerydestination.spec:14]: should be named `isIndirect`

**Review by primalmotion — CHANGES_REQUESTED** [review]

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "b4ad721b8f1c5ca55f9cdf85a8c10cd9e674b78a",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "e93cae51f092e06c71a180938f93c99847252f47",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "cd10ef611d4c02d374e95059e953bb97d5a484aa",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "cd10ef611d4c02d374e95059e953bb97d5a484aa",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "cd10ef611d4c02d374e95059e953bb97d5a484aa",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1106",
    "commit-sha": "d403e6bab5ae4638360337eaa473f2e7a546820e",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1112",
    "commit-sha": "32ede09a65fcd251754bdc4223f47cebfd1cbb8f",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  }
]
```

**Comment by dstiliadis** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "graph-services",
    "component": "backend",
    "pr-id": "1112",
    "commit-sha": "976bc7c877aa91bd181065c84252026f4af9f568",
    "pipeline": "master"
  },
  {
    "project": "graph-services",
    "component": "gaia",
    "pr-id": "632",
    "commit-sha": "cc2f6945daf7ba34aeb8e8895e95edf627c94fd7",
    "pipeline": "master"
  }
]
```

**Pull request merged**